Makefile

```diff
@@ -442,6 +442,50 @@ $(BINDIR)/gen-versions: $(shell find ./hack/gen-versions -type f)
 	sh -c '$(GIT_CONFIG_SSH) \
 	go build -o $(BINDIR)/gen-versions ./hack/gen-versions'
 
+## Generate a ClusterServiceVersion package.
+# E.g. make gen-csv VERSION=1.6.2 PREV_VERSION=0.0.0
+#
+# VERSION: the operator version to generate a CSV for.
+# PREV_VERSION: the operator version that this CSV will replace. If there is
+#               no previous version, use 0.0.0
+.PHONY: gen-csv
+gen-csv: hack/bin/operator-sdk-$(OPERATOR_SDK_VERSION) get-digest
+	@echo Operator image digest is $(OPERATOR_IMAGE_DIGEST)
+	$(eval EXTRA_DOCKER_ARGS += -e OPERATOR_IMAGE_DIGEST="$(OPERATOR_IMAGE_DIGEST)" -e VERSION=$(VERSION) -e PREV_VERSION=$(PREV_VERSION))
+	$(CONTAINERIZED) hack/gen-csv/csv.sh
+
+.PHONY: prepull-image
+prepull-image:
+	@echo Pulling operator image...
+	docker pull quay.io/tigera/operator:v$(VERSION)
+
+# Get the digest for the image. 'docker inspect' returns output like the example
+# below. RepoDigests may have more than one entry so we need to filter.
+# [
+#     {
+#         "Id": "sha256:34a1114040c03830da0a8d57f8d999deba26d8e31bda353aed201a375f68870b",
+#         "RepoTags": [
+#             "quay.io/tigera/operator:v1.3.1",
+#             "..."
+#         ],
+#         "RepoDigests": [
+#             "quay.io/tigera/operator@sha256:5e1d551b5a711592472f4a3cc4645698d5f826da4253f0d47cfa5d5b641a2e1a",
+#             "..."
+#         ],
+#         ...
+#     }
+# ]
+.PHONY: get-digest
+get-digest: prepull-image
+	@echo Getting operator image digest...
+	$(eval OPERATOR_IMAGE_DIGEST=$(shell sh -c "docker image inspect quay.io/tigera/operator:v$(VERSION) | jq -r '.[0].RepoDigests[] | select(. | contains(\"quay.io/tigera/operator\"))' "))
```

Comment:
This is better but I still don't like depending on having jq on the host system.
It would be easy enough to just pass in the whole inspect output right?

Reply:
Fixed. I had to base64 encode it to get around escaping issues.